app/http/requests/login/login_request.go

```diff
@@ -0,0 +1,67 @@
+package login
+
+import (
+	"github.com/goravel/framework/contracts/http"
+	"github.com/goravel/framework/contracts/validation"
+)
+
+type Captcha struct {
+	ID   string `form:"id" json:"id"`
+	Code string `form:"code" json:"code"`
+}
+
+type LoginRequest struct {
+	Username string  `form:"username" json:"username"`
+	Password string  `form:"password" json:"password"`
+	Captcha  Captcha `form:"captcha" json:"captcha"`
+	Origin   string  `form:"origin" json:"origin"`
+}
+
+func (r *LoginRequest) Authorize(ctx http.Context) error {
+	return nil
+}
+
+func (r *LoginRequest) Filters(ctx http.Context) map[string]string {
+	return map[string]string{
+		"username": "trim",
+		"origin":   "trim",
+	}
+}
+
+func (r *LoginRequest) Rules(ctx http.Context) map[string]string {
+	return map[string]string{
+		"username":     "required|string|min:1|max:100",
+		"password":     "required|string|min:5",
+		"captcha.id":   "required|string",
+		"captcha.code": "required|string",
+		"origin":       "required|string|max:100",
```

Comment:
`min` and `max` are for the number type, please use `min_len` and `max_len` here.

```suggestion
		"username":     "required|string|min_len:1|max_len:100",
		"password":     "required|string|min_len:5",
		"captcha.id":   "required|string",
		"captcha.code": "required|string",
		"origin":       "required|string|max_len:100",
```